src/smt/theory_str_noodler/formula.h

```diff
@@ -693,13 +748,37 @@ namespace smt::noodler {
             return "to_int";
         case ConversionType::FROM_INT:
             return "from_int";
-        
+
         default:
             UNREACHABLE();
             return "";
         }
     }
 
+    // ***** Convert noodler len nodes to z3 expressions
+    /**
+     * Context (other formulae) in which a formula is to be used.
+     */
+    struct LenFormulaContext {
+        ast_manager& manager;
+        arith_util&  arith_utilities;
+        seq_util  &  seq_utilities;
+
+        /**
+         * Maps quantified variable names to the quantifier-depth of the corresponding quantifier.
+         *
+         * Necessary to correctly compute de Brujin sequences for the indices, which facilitate
+         * the binding between a quantifier and a new variable.
+         */
+        std::map<std::string, unsigned>& quantified_vars;
+        std::map<BasicTerm, expr_ref>&   known_z3_exprs;
+
+        int current_quantif_depth = 0;  // To compute de Brujin numbers for quantified variables
+    };
+    expr_ref convert_len_node_to_z3_formula(LenFormulaContext &ctx, const LenNode &node);
```

Comment:
Do we now have two functions that transform len node to z3 formula? We should delete the other one then.

Reply:
Yes, we have two functions - but one is incorrect in its handling of quantified variables. I will remove the incorrect one.

Reply:
Can I mark it as resolved? I noticed you removed it.